### ISSUE TITLE
[CUDA 13.2] Inductor training accuracy regression for BertForMaskedLM (HF benchmark) vs CUDA 13.0

### ISSUE THREAD

**Issue by atalman** [pytorch/pytorch#191037]
### Summary

On the new CUDA 13.2 inductor CI lane, the `inductor_huggingface` **training** accuracy check regresses for **`BertForMaskedLM`**, while the same model/suite passes on the existing CUDA 13.0 lane (same A10G / sm86 runner, same commit).

`check_accuracy.py` verdict on cu13.2:
```
Error: 1 models have accuracy status regressed:
    BertForMaskedLM
```

### Details

The failing comparison is the model's **scalar training loss** (`shape=torch.Size([])`, `numel=1`), so the fp64 RMSE ratio is noise-dominated:

```
RMSE (res-fp64): 0.10322, (ref-fp64): 0.03272, shape=torch.Size([]),
res.dtype: torch.float32, multiplier: 3.000000, tol: 0.010000
```

The accuracy gate is `res_error <= multiplier * ref_error + tol/10`:
- cu13.2: `0.10322 <= 3.0 * 0.03272 + 0.001 = 0.09916` → **FAIL** (over by ~0.004)
- cu13.0: same model → **PASS** (res_error is below the 3x threshold)

So `res_error` (compiled-vs-fp64) is ~3.15x the eager `ref_error` (eager-vs-fp64) on CUDA 13.2 but within 3x on CUDA 13.0. For a scalar loss this is close to noise, but it is a reproducible cu13.2-vs-cu13.0 delta — most likely a cuBLAS/cuSOLVER numeric change in the 13.2 toolkit.

### Evidence

- Failing (cu13.2): https://github.com/pytorch/pytorch/actions/runs/30049239912/job/89362543600
- Passing (cu13.0, same suite/runner, `main`): https://github.com/pytorch/pytorch/actions/runs/30057644058/job/89375694344
- The 5 `eager_fail_to_run` models (Blenderbot, Qwen3-0.6B, gemma-2-2b, Llama-3.2-1B, whisper-tiny) are A10G OOMs and are **expected/tolerated on both lanes** — they are not the regression.

### Workaround (already applied to unblock the cu13.2 lane)

`BertForMaskedLM` was added to `use_larger_multiplier_for_smaller_tensor` in `benchmarks/dynamo/huggingface.py` (as already done for `GPT2ForSequenceClassification`), granting a 10x multiplier for the scalar output (threshold → 0.3282), which passes on cu13.2 and does not affect cu13.0 (already passes at the stricter 3x). Landed in #190948.

**This issue tracks the underlying numeric divergence**, which the tolerance bump masks. If the cu13.2 delta is confirmed benign (scalar-loss noise), this can be closed; if `res_error` grows further or appears on larger-tensor outputs, it should be root-caused in the 13.2 math libraries.

### Environment

- CUDA 13.2 (vs 13.0 baseline), inductor, training, `inductor_huggingface` suite
- Runner: `mt-l-x86aavx2-29-113-a10g` (A10G, sm86), float32
- Model: `BertForMaskedLM`

### Repro

```
python benchmarks/dynamo/huggingface.py --accuracy --training --backend inductor \
  --device cuda --only BertForMaskedLM
```

cc @ptrblck @msaroufim @eqy @tinglvv @nWEIdia @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo
